cmd/containerd-shim-runhcs-v1/pod.go

```diff
@@ -262,17 +258,10 @@ func (p *pod) CreateTask(ctx context.Context, req *task.CreateTaskRequest, s *sp
 		return nil, errors.Wrapf(errdefs.ErrFailedPrecondition, "task with id: '%s' cannot be created in pod: '%s' which is not running", req.ID, p.id)
 	}
 
-	p.wcl.Lock()
-	_, loaded := p.workloadTasks.LoadOrStore(req.ID, nil)
-	if loaded {
+	_, ok := p.workloadTasks.Load(req.ID)
+	if ok {
 		return nil, errors.Wrapf(errdefs.ErrAlreadyExists, "task with id: '%s' already exists id pod: '%s'", req.ID, p.id)
 	}
-	p.wcl.Unlock()
-	defer func() {
-		if err != nil {
-			p.workloadTasks.Delete(req.ID)
-		}
```

Comment:
do we not want this anymore?

Reply:
This was only to handle the case when the `LoadOrStore` call above stored a `nil` entry into the map (because the entry with key `req.ID` doesn't already exist) but then we fail somewhere after this point. We don't want to leave the `nil` entry in the map in that case. Now we use `Load` instead of `LoadOrStore` so we won't add an entry into the map at all unless container creation is successful so we don't need this defer anymore.